goloose.go

```diff
@@ -9,13 +9,29 @@ import (
 	"time"
 )
 
+type Options struct {
+	StringToFloat64 bool // controls whether goloose will convert strings to floats if required; note this breaks the json.Unmarshal paradigm
```

Comment:
I wonder if this option should be more generic than this: for instance, I could see wanting to be able to convert the string `"true"` to a bool, an int to a string, etc. Maybe something like `CoerceStrings`?

Reply:
Note that what I'm describing is a bunch more work though, so if you don't have time for it I'd be happy to just take this as-is and stick it on my backlog of things to do eventually :)

Reply:
Sorry, totally missed these comments! Bool's in there, but I'll make a PR for adding in the other numeric types.